### DISCUSSION TITLE
How to have a common `ocelot.json` configuration on DFS?

### DISCUSSION THREAD

**Discussion by stamatiagk** [ThreeMammals/Ocelot#2105] [Q&A] [answered]
## Expected Behavior / New Feature
When the gateway is installed in multiple servers with distributed file system (DFS) and in front of those servers a load balancer exists that manages the traffic, the **ocelot.json** needs write permission in each server and when updating it with the admin API it ends up having different files in each server.
- How it is possible to have a common **ocelot.json** in this case?

### Message from repo maintainer on November 30, 2023
Hi lady!
Your this question-issue has been answered by @TomPallister and @PKompis. 
Here is the answer by @PKompis [on Oct 16, 2018](https://github.com/ThreeMammals/Ocelot/discussions/2105#discussioncomment-9781972)

I hope, this question-issue can be converted to true issue which is ready for development, if we will have active contributor.
Are you going to contribute?...

**Comment by TomPallister** [comment]
@stamatiagk hey, someone needs to implement IFileConfigurationRepository for some kind of shared storage e.g. database or you can store your configuration in consul. You can see how that has been done [here](https://github.com/ThreeMammals/Ocelot.Provider.Consul). I would suggest just implementing it for yourself and see if you can get it working. Then we get can get a PR afterwards.

**Comment by PKompis** [comment]
Hi @TomPallister, together with @stamatiagk we implemented an `IFileConfigurationRepository` for an SQL Server using Dapper. 
In order to implement it we made two changes.
1. In Ocelot package, if there is any `IDbConnection` injected then the new `IFileConfigurationRepository` implementation is used.
2. In `IFileConfigurationRepository` interface `Set` method a parameter named `isAdministrativeCall` added in order to distinguish if the `Set` method is called from Admin Post API Call.

What do you think ? Any recommendation? I thought I should ask you before I create a PR.

**Comment by raman-m** [comment]
> @PKompis [on Oct 16, 2018](https://github.com/ThreeMammals/Ocelot/discussions/2105#discussioncomment-9781972)

Hi Panagiotis!
Sorry, we've missed 5 years on this thing!...
Your recipe with config saving to DB is good.
Did you save this feature branch or have some Ocelot repo copy on HDD? I don't see Ocelot fork in [your repos](https://github.com/PKompis?tab=repositories)...
Are you still with Ocelot?

----
> What do you think ? Any recommendation? I thought I should ask you before I create a PR.

If you'll reply then I'll share you details on dev plan...

**Comment by raman-m** [comment, marked as answer]
@stamatiagk 
> How it is possible to have a common ocelot.json in this case?

You need to build one Docker image with Ocelot config inside and create Docker containers by Kubernetes or other container tools.

### 2nd design
If you want to use DFS for all Ocelot instances (ignoring current Ocelot instance file system) then Ocelot doesn't have such functionality to reuse DFS interfaces. It must be developed.
